common/metrics/tally/statsd/reporter_test.go

```diff
@@ -60,3 +62,35 @@ func TestMetricNameWithTagsStability(t *testing.T) {
 		assert.Equal(t, r.metricNameWithTags(name, tags), r.metricNameWithTags(name, tags))
 	}
 }
+
+func TestMetricNameWithSeparatedTags(t *testing.T) {
+	testCases := []string{
```

Comment:
Can you add the empty value case?
That would cover regressions where this configuration entry is not present.